### PR TITLE
boards: suppress dtc warning for SPI bridge on b_u585i_iot02a

### PR DESCRIPTION
**Pull request by toonst**
As discussed [here](https://github.com/zephyrproject-rtos/zephyr/issues/24499), the current way to suppress DTC warnings is through `pre_dt_board.cmake`.

### CODE REVIEW

File: boards/arm/b_u585i_iot02a/pre_dt_board.cmake
Comment (kartben) [inline boards/arm/b_u585i_iot02a/pre_dt_board.cmake:2]: I would use a descriptive comment like for other boards 

```suggestion
# SPI is implemented via octospi so node name isn't spi@...
```

Comment (kartben) [reply boards/arm/b_u585i_iot02a/pre_dt_board.cmake:2]: In fact, it would probably make sense to update all the other boards that have a similar issue (stm32h735g_disco, stm32h573i_dk, stm32h7b3i_dk, stm32l562e_dk, ...) but I am not sure what's an easy way to list them all.

**Review by kartben — CHANGES_REQUESTED** [review]

File: boards/arm/b_u585i_iot02a/pre_dt_board.cmake
Comment (kartben) [reply boards/arm/b_u585i_iot02a/pre_dt_board.cmake:2]: To clarify: my request for change is only about making the comment more useful, not about asking you to do the updates for all boards using octospi :)

Comment (toonst) [reply boards/arm/b_u585i_iot02a/pre_dt_board.cmake]: Instead of adding a `pre_dt_board.cmake` for all of these boards, it might make sense to suppress the warning on a global level for all boards? Is there a benefit for having this warning in certain cases?

**Review by kartben — APPROVED** [review]

**Comment by kartben** [comment]
@toonst please address compliance errors. Sorry that I let this PR go stale :/

**Comment by toonst** [comment]
@kartben sorry for the oversight on my part as well, done now. Still wondering if it makes more sense to suppress the warnings on a global level (for all boards). Maybe there could be a `pre_dt.cmake` in the boards directory that contains global warning suppresions?

**Review by erwango — CHANGES_REQUESTED** [review]
This warning could be seen on a variety of boards and will be visible on out of tree boards as well. I think this deserves a more global approach.

Maybe device tree maintainers can suggest a better solution. @mbolivar-ampere @galak

**Comment by kartben** [comment]
@erwango could you maybe open a different issue for this? I am not sure it makes sense to block this one from going in until we have a more global solution?

**Review by erwango — APPROVED** [review]

**Comment by erwango** [comment]
> Ok. will do

It seems it is actually sufficiently described in https://github.com/zephyrproject-rtos/zephyr/issues/24499. So, I won't open a new issue. 
Also, according to https://github.com/zephyrproject-rtos/zephyr/issues/24499#issuecomment-1696131170, there is not way to solve this soc wide. This being said fixing it in a board just seems a loss of time to me.

**Comment by dleach02** [comment]
The GitHub GUI is weird right now. It shows two approvals (@erwango and @kartben) but when I view the list of reviewers at the top right it only shows @kartben?

**Comment by spectrum70** [comment]
Hi, i am getting the same warning, will use the same solution as in the commit.

**Pull request merged**
